### ISSUE TITLE
Error Compiling Cache_Template_Attacks

### ISSUE THREAD

**Issue by pg2527** [IAIK/armageddon#4]
Works fine for ARCH=x86 but when I change the architecture to ARCH=armv7 and WITH_ANDROID=1 , it fails to compile saying that " libflush.h" not found"

**Comment by psyte** [comment]
First make sure that libflush and (its example) compiles correctly for armv7.
Btw I don't believe that WITH_ANDROID=1 is required if you haven't touched the makefiles.
Then make sure that the source-directory in your config.yml points to where your libflush Makefile lives.

**Comment by pg2527** [comment]
Libflush compiles for armv7 without any error. It is mentioned in the instructions that we need to use WITH_ANDROID option if it is build for an android device. Also why are you talking about config.yml here? That is a part of eviction_strategy_evaluator. Correct me if I am wrong..

**Comment by psyte** [comment]
Ok my fault, I thought you were talking about the evaluator. I haven't tried building the cache_template_attacks project yet. I guess you have "sudo make install"ed libflush..?

**Comment by pg2527** [comment]
Yes.

**Comment by psyte** [comment]
So I just tried and it builds for me. But I'm building for armv8 using the vanilla gcc toolchain directly on my devboard (not cross-compiling).
Check that your {TARGET}/usr/include/libflush/libflush.h is installed and that your ANDROID_NDK_PATH is configured in config-arm.mk

Sorry I couldn't help more.

**Comment by pg2527** [comment]
NDK path is set correctly. and yeah I have libflush.h in my include folder. I will try to build it using android-ndk.

**Comment by pg2527** [comment]
I could finally build it , but on running it, I encountered an error 
![libflush_error](https://cloud.githubusercontent.com/assets/17165273/20938212/7702ad2c-bbea-11e6-80eb-af6aa68bfa07.png)
Could you please look into this

**Comment by mlq** [comment]
Did you manage to build it in the mean time? As you say that it works for x86 I assume that ``LIBFLUSH_INC`` and ``LIBFLUSH_LIB`` is set correctly. In addition you can pass ``VERBOSE=1`` to make to retrieve a more verbose build output that might help to track down this issue.

**Comment by pg2527** [comment]
Yes I managed to build it for armv7 by changing LIBFLUSH_INC and LIBFLUSH_LIB in config.mk.. However, the default config.mk works fine for x86 architectures. I guess there are some addressing issues which are needed to be fixed. Lately, I ran into another issue, when I tried to run these attacks on Nexus 5, they fail stating below mentioned error
![libflush_error](https://cloud.githubusercontent.com/assets/17165273/20949933/87ce2190-bc1c-11e6-8147-508ab3cb7c72.png)
I also tried to run these attacks on x86, they work fine but not on armv7 based device, they halt.

**Comment by mlq** [comment]
This means that the library was not able to find as many addresses that map to the same cache set as the eviction strategy needs. What you can do is to increase the amount of memory that is used to find those addresses. You can do that by changing ``PHYSICAL_MEMORY_MAPPED_SIZE`` in ``./libflush/eviction/configuration.h``.

**Comment by psyte** [comment]
Also make sure that you have access to /proc/{pid]/pagemap (i.e. sudo/root), sine-qua-non you might get this same error.

**Comment by dgruss** [comment]
that's a recent patch... when we developed the attack it was still 
accessible... an assertion might be useful...

**Comment by pg2527** [comment]
Yeah, root permission is needed to access /proc/{pid}/pagemap . Also I tried to increase the PHYSICAL_MEMORY_MAPPED_SIZE but nothing is happening. Is there any limit upto which I should do that ( upto the size of RAM or so). It is now not throwing any error but it seems , it is stuck. @psyte did you run these attacks on your devboard.
